Fix get taxonomy in a context without request

tdesvenain opened the pull request.

vincentfretin commented on src/collective/taxonomy/interfaces.py:
```diff
@@ -52,16 +52,26 @@ class ITaxonomySettings(Interface):
+        if lang is None:
+            lang = api.portal.get_current_language()
+        return lang.split('-', 1)[0]
+    except AttributeError:
```
Why do you get an AttributeError?
Are you sure you have a recent version of plone.api?

tdesvenain commented:
with plone.api 1.6.0

  File "/DATA/mpi.si/src/mpi/si/utils.py", line 291, in get_vocabulary
    return getUtility(IVocabularyFactory,
name=factory_name)(api.portal.get())
  File "/DATA/
mpi.si/src/collective.taxonomy/src/collective/taxonomy/utility.py", line
53, in __call__
    current_language = self.getCurrentLanguage(request)
  File "/DATA/
mpi.si/src/collective.taxonomy/src/collective/taxonomy/utility.py", line
79, in getCurrentLanguage
    language = get_lang_code()
  File "/DATA/
mpi.si/src/collective.taxonomy/src/collective/taxonomy/interfaces.py", line
63, in get_lang_code
    lang = api.portal.get_current_language()
  File "/DATA/.buildout/eggs/plone.api-1.6-py2.7.egg/plone/api/portal.py",
line 420, in get_current_language
    return request.get('LANGUAGE', None) or \
AttributeError: 'NoneType' object has no attribute 'get'


2017-05-18 16:59 GMT+02:00 Vincent Fretin <notifications@github.com>:

> *@vincentfretin* commented on this pull request.
> ------------------------------
>
> In src/collective/taxonomy/interfaces.py
> <https://github.com/collective/collective.taxonomy/pull/41#discussion_r117271872>
> :
>
> >      if not taxonomy:
>          return u''
>      return taxonomy
>
>
>  def get_lang_code(lang=None):
> -    if lang is None:
> -        lang = api.portal.get_current_language()
> -    return lang.split('-', 1)[0]
> +    try:
> +        if lang is None:
> +            lang = api.portal.get_current_language()
> +        return lang.split('-', 1)[0]
> +    except AttributeError:
>
> Why do you get an AttributeError?
> Are you sure you have a recent version of plone.api?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/collective/collective.taxonomy/pull/41#pullrequestreview-38964663>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAU39xZ0rfR4gnDuV33bI9Zp28pchVvDks5r7FzPgaJpZM4NfVOo>
> .
>



-- 
Thomas Desvenain

Téléphone : 09 51 37 35 18

vincentfretin commented:
You get an AttributeError because the request is None. A better solution would be to check if you have a request in getCurrentLanguage and use the right API.
